api/src/HgRunner.php

```diff
@@ -58,30 +54,60 @@ function unbundle($filepath) {
 		// run hg incoming to make sure this bundle is related to the repo
 		$cmd = "hg incoming $filepath";
 		$this->logEvent("cmd: $cmd");
-		$asyncRunner = new AsyncRunner($filepath . ".incoming");
+		$asyncRunner = $this->getValidationRunner($filepath);
 		$asyncRunner->run($cmd);
-		$asyncRunner->synchronize();
-		$output = $asyncRunner->getOutput();
-		if (preg_match('/abort:.*unknown parent/', $output)) {
-			throw new UnrelatedRepoException("Project is unrelated!  (unrelated bundle pushed to repo)");
-		}
-		if (preg_match('/parent:\s*-1:/', $output)) {
-			throw new UnrelatedRepoException("Project is unrelated!  (unrelated bundle pushed to repo)");
+	}
+
+	/**
+	 * @param AsyncRunner $asyncRunner
```

Comment:
```suggestion
	 * @param string $filePath
```
fix up documentation

Reply:
Fixed